src/jdk/src/solaris/native/java/lang/java_props_macosx.c

```diff
@@ -185,11 +185,16 @@ void setOSNameAndVersion(java_props_t *sprops) {
 
     char* osVersionCStr = NULL;
     // Mac OS 10.9 includes the [NSProcessInfo operatingSystemVersion] function,
-    // but it's not in the 10.9 SDK.  So, call it via objc_msgSend_stret.
+    // but it's not in the 10.9 SDK.  So, call it via NSInvocation.
```

Comment:
You might have to conditionalize this code on the OS version. Early versions may not support NSInvocation.

Reply:
Per [NSInvocation](https://developer.apple.com/documentation/foundation/nsinvocation) docs, available from 10.0+. So I think this change is safe

Reply:
Ok